pkg/constants/constants.go

```diff
@@ -14,14 +14,26 @@
 
 package constants
 
+import "os"
+
 const (
 	CurrentAPIVersion = "krew.googlecontainertools.github.com/v1alpha2"
 	PluginKind        = "Plugin"
 	ManifestExtension = ".yaml"
 	KrewPluginName    = "krew" // plugin name of krew itself
 
-	// DefaultIndexURI points to the upstream index.
-	DefaultIndexURI = "https://github.com/kubernetes-sigs/krew-index.git"
 	// DefaultIndexName is a magic string that's used for a plugin name specified without an index.
 	DefaultIndexName = "default"
+
+	// defaultIndexURI is the default krew-index URI that gets used unless it's overridden
+	defaultIndexURI = "https://github.com/kubernetes-sigs/krew-index.git"
 )
+
+// DefaultIndexURI points to the upstream index.
+var DefaultIndexURI = defaultIndexURI
```

Comment:
sorry for the after-the-merge review.

I think we should not have code in pkg/constants that are actually not constants and change in runtime.

I recommend keeping this package as-is and implementing a DefaultIndex() method in another package, like pkg/index.

Reply:
no worries, that makes sense. I think I can get a PR for that in later today